backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -1293,4 +1301,323 @@ private Boolean blobTypeExistsInList(List<BlobType> blobTypeList, Blob.ContentCa
       }
     }
   }
+
+  @Override
+  public GetUrlForBlobVersioned.Response getUrlForVersionedBlob(
+      ArtifactStoreDAO artifactStoreDAO,
+      RepositoryFunction repositoryFunction,
+      CommitFunction commitFunction,
+      GetUrlForBlobVersioned request)
+      throws ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      RepositoryEntity repositoryEntity = repositoryFunction.apply(session);
+      CommitEntity commitEntity = commitFunction.apply(session, session1 -> repositoryEntity);
+
+      Map<String, Map.Entry<BlobExpanded, String>> locationBlobWithHashMap =
+          getCommitBlobMapWithHash(
+              session,
+              commitEntity.getRootSha(),
+              request.getLocationList(),
+              Collections.singletonList(BlobType.DATASET_BLOB));
+      String presignedUrl =
```

Comment:
could we just pass the value to the function instead of entire HashMap? This was we can also pull in the validation sooner and keep the function more generic

Reply:
Fixed as per comment